File: api/spring-boot-api/src/main/java/com/spicytodo/dto/TodoUpdateRequest.java
package com.spicytodo.dto;

import com.spicytodo.model.Priority;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class TodoUpdateRequest {
    @Size(min = 1, max = 500, message = "Todo text must be between 1 and 500 characters")
    private String text;
    
    private Priority priority;
    private Boolean completed;
    private String dueDate;
    private String reminderTime;
}

File: api/dotnet-api/Models/TodoUpdateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace DotnetApi.Models
{
    public class TodoUpdateRequest
    {
        [StringLength(500, MinimumLength = 1, ErrorMessage = "Todo text must be between 1 and 500 characters")]
        public string? Text { get; set; }
        
        public Priority? Priority { get; set; }
        public bool? Completed { get; set; }
        public string? DueDate { get; set; }
        public string? ReminderTime { get; set; }
    }
}

